src/rewards/community_advisors/lottery.rs

```diff
@@ -26,13 +26,16 @@ pub fn lottery_distribution<R: Rng>(
 
     // Consistent iteration is needed to get reproducible results. In this case,
     // it's ensured by the use of BTreeMap::iter()
-    for (ca, n_tickets) in distribution.into_iter() {
+    for (ca, n_tickets) in distribution.iter_mut() {
         let tickets_won = std::iter::from_fn(|| {
-            indexes.next_if(|tkt| *tkt < (cumulative_ticket_index + n_tickets) as usize)
+            indexes.next_if(|tkt| *tkt < (cumulative_ticket_index + *n_tickets) as usize)
         })
         .count();
-        cumulative_ticket_index += n_tickets;
-        winnings.insert(ca, tickets_won as u64);
+        cumulative_ticket_index += *n_tickets;
+        if tickets_won > 0 {
+            winnings.insert(ca.clone(), tickets_won as u64);
```

Comment:
Conversion from `usize` to `u64` is potentially dangerous. I would not rely on the assumption of having 64-bit `usize`.

Reply:
Why? If something I would expect the opposite to be more dangerous, don't expect 128 bits archs to become that common in the near future.
Also that `usize` is actually bound to the limits of `u64` as the total number of tickets is a `u64`.

Reply:
I'm talking about the possibility of running it on a 32-bit system :)

Reply:
Ah well, I was dumb, it would break with the opposite conversion. Sorry.